invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/Modals.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2022 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React from "react";
+import { i18next } from "@translations/invenio_communities/i18next";
+import { Button, Modal } from "semantic-ui-react";
+import { ErrorMessage } from "./ErrorMessage";
+
+export const LeaveCommunityModal = ({
+  loading,
+  error,
+  confirmAction,
+  open,
+  cancelAction,
+  onClose,
```

Comment:
why this modal is not managing its own state but passes the responsibility to the parent ? it will have always only one instance, because there is only one current user who could leave the community

Reply:
yes but you still have to include the modal for every item even if only one user can trigger it, which we are trying to avoid, right?